data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,308 @@
+
+## Using the Preprocessor to Generate Code
+
+_FIXME: Shall we move this to another tutorial? Which one?_
+
+Let's assume we'd like `hello` to display its output as if it was a list of strings in UTop: `["hello"; "using"; "an"; "opam"; "library"]`. To do that, we need a function turning a `string list` into a `string`, adding brackets, spaces, and commas. Instead of defining it ourselves, let's generate it automatically with a package. We'll use `ppx_show`, which was written by Thierry Martinez. Here is how to install it:
+```shell
+$ opam install ppx_show
+```
+
+Dune needs to be explained that we are using it. Edit the `lib/dune` file to look like this:
+```lisp
+(library
+ (name hello)
+ (preprocess (pps ppx_show))
+ (libraries ppx_show.runtime))
+```
+
+Here is the meaning of the two new lines:
+- `(libraries ppx_show.runtime)` means our project is using definitions found in the `ppx_show.runtime` library, provided by the package `ppx_show`; 
+- `(preprocess (pps preprocess))` means that before compilation the source needs to be transformed using the preprocessor provided by the package `ppx_show`.
+
+The files `lib/hello.ml` and `lib/hello.mli` need to be edited, too:
+
+##### `lib/hello.mli`
+```ocaml
+val string_of_string_list : string list -> string
+val world : string list
+
+```
+##### `lib/hello.ml`
+```ocaml
+let string_list_pp = [%show: string list]
+
+let string_of_string_list = Format.asprintf "@[%a@]" string_list_pp
+
+let world = String.split_on_char ' ' "Hello using an opam library"
+```
+
+Let's read this from the bottom up:
+- `world` has the type `string list`. We're using `String.split_on_char` to turn a `string` into a `string list` by splitting the string on space characters.
+- `string_of_string_list` has type `string list -> string`. This converts a list of strings into a string, applying the expected formatting. 
+- `string_list_pp` has type `Format.formatter -> string list -> unit`, which means it is a custom formatter that turns a `string list` into a `string` (this type does not appear in the signature).
+
+Finally, you'll also need to edit `bin/main.ml`
+```ocaml
+let () = print_endline Hello.(string_of_string_list world)
+```
+
+Here is the result:
+```shell
+dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Done: 90% (19/21, 2 left) (jobs: 1)["Hello"; "using"; "an"; "opam"; "library"]
+```
+
+## A Sneak-Peek at Dune as a One-Stop Shop
+
+This section explains the purpose of the files and folders created by `dune proj init` which haven't been mentioned earlier.
+
+Along its history, several tools have been used. As of writing this tutorial (Summer 2023), Dune is the mainstream one, which is why it is used in the tutorial. Dune automatically extracts the dependencies between the modules from the files and compiles them in a compatible order. It only needs one `dune` file per folder where there is something to build. The three folders created by `dune proj init` have the following purposes:
```

Comment:
Dune is the mainstream "tool" or "build system"?
Mainstream meaning most people use it?
... or should that be "main" as in the primary build system?

Reply:
Not 100% sure it is the build system most people use. But among build systems for OCaml, this is the one which is most used. Relative majority is a safe bet; absolute majority is not proved. 

Is mainstream ok in this context